### ISSUE TITLE
Crouching and jumping not working right

### ISSUE THREAD

**Issue by zach-lawrence-1** [Brackeys/2D-Animation#3]
Crouching is constantly being turned on and off and the animation is looping from crouching to idle over and over. Also jumping animation doesn't play unless I'm already in the air and press space when it should activate when I am not touching the ground.

**Comment by imprisonedmind** [comment]
Did you ever resolve this?
I have the same issue

**Comment by zach-lawrence-1** [comment]
Yes, but I can't remember how because it has been almost a year. I'll try to look at my old project if you want. Reply back if you want me to look at it

**Comment by imprisonedmind** [comment]
I managed to find a fix in one of the pull requests. 
Thanks for the reply.

**Comment by Ne9nX** [comment]
How did you fix the Crouching animation?

**Comment by imprisonedmind** [comment]
I think I looked through the pull requests and found a change at the code, Sorry it was long ago I am not sure now

**Comment by AlfrogKun** [comment]
Has anyone found the actual solution t o this yet? if so, could you please link it, thanks!

**Comment by hran-8** [comment]
Line 67, delete the ! from `if (!crouch)`

**Comment by nullStringTerminator** [comment]
i have got a problem with the jumping.

crouching is fine but it will only play my jump animation fully when i press space while in the air;
however my jump animation is visible for several frames when space is pressed while on the ground before returning to my run, or idle animation.

my guess is that it's probably detecting a collision before fully clearing the floor therefore triggering this bit of code: public void 
`public void OnLanding ()
	{
		animator.SetBool("jumping", false);
	}`
early!

**Comment by nullStringTerminator** [comment]
however I have spent the good part of two and a half hours trying to fix it but have failed!

I am new to unity and don't know if there is a simple fix to this.
